src/python.rs

```diff
@@ -0,0 +1,31 @@
+use std::io::Cursor;
+
+use crate::{error::Error, eval::cache::CBNCache, program::Program};
+
+use pyo3::{create_exception, exceptions::PyException, prelude::*};
+
+create_exception!(nickel_lang, NickelException, PyException);
+
+impl std::convert::From<Error> for PyErr {
+    fn from(err: Error) -> PyErr {
+        match err {
+            // TODO better exceptions
+            error => NickelException::new_err(format!("unexpected error: {error:?}")),
+        }
+    }
+}
+
+// Some test function
+#[pyfunction]
+pub fn run(s: String) -> PyResult<String> {
+    let mut program: Program<CBNCache> =
```

Comment:
FYI, if you want to serialize to JSON instead, here is the code of the main binary that does that: https://github.com/tweag/nickel/blob/bff8451f7e8e860859070e6f4f10221f79bef1f2/src/bin/nickel.rs#L222.

There is not much more: use `eval_full()` instead of `eval` (which substitutes the strange `%123` variables for their actual value), valide the result (check if there isn't any function or other non-serializable expression remaining), and then use `serialize::to_xxx`. In particular there is a `serialize::to_string` function.

Reply:
Thanks a lot! I could implement it in a454f9a1 💯